CLOUDP-200398: unhide deployments commands (#2263)
mongodb/mongodb-atlas-cli: docs/atlascli/command/atlas-deployments-setup.txt

Author: pending for review

Reviewer: LGTM!

.. _atlas-deployments-setup:

=======================
atlas deployments setup
=======================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

Create a local deployment.

Syntax
------

.. code-block::
   :caption: Command Syntax

   atlas deployments setup [deploymentName] [options]

.. Code end marker, please don't delete this comment

Arguments
---------

.. list-table::
   :header-rows: 1
   :widths: 20 10 10 60

   * - Name
     - Type
     - Required
     - Description
   * - deploymentName
     - string
     - false

>      - Name of the deployment that you want to setup.

Reviewer: ```suggestion
     - Name of the deployment that you want to set up.
```

Author: done, thanks

Options
-------

.. list-table::
   :header-rows: 1
   :widths: 20 10 10 60

   * - Name
     - Type
     - Required
     - Description
   * - --accessListIp
     - strings
     - false

>      - IP address to be allowed to access the deployment.

Reviewer: ```suggestion
     - IP address to grant access to the deployment.
```

Author: done, thanks

       Mutually exclusive with --currentIp.
   * - --connectWith
     - string
     - false
     - Method for connecting to the deployment. Valid values are mongosh, compass, and skip.
   * - --currentIp
     - 
     - false
     - Flag that adds the IP address from the host that is currently executing the command to the access list.

       Mutually exclusive with --accessListIp.
   * - --enableTerminationProtection
     - 
     - false

>      - Enables termination protection for your cluster. You can't delete a cluster with termination protection enabled.

Reviewer: ```suggestion
     - Enables termination protection for your deployment. You can't delete a deployment with termination protection enabled.
```

Author: done, thank you

   * - --force
     - 
     - false

>      - If specified, skips asking for input and creates a cluster with default settings for any options you don't specify.

Reviewer: ```suggestion
     - If specified, skips asking for input and creates a deployment with default settings for any options you don't specify.
```

Author: done, thank you

   * - -h, --help
     - 
     - false
     - help for setup
   * - --mdbVersion
     - string
     - false

>      - Major MongoDB version of the cluster.

Author: done, thank you

   * - --password
     - string
     - false
     - Password for the user.
   * - --port
     - int
     - false
     - Port that the MongoDB server listens to for client connections.
   * - --projectId
     - string
     - false
     - Hexadecimal string that identifies the project to use. This option overrides the settings in the configuration file or environment variable.
   * - --provider
     - string
     - false
     - Name of your cloud service provider. Valid values are AWS, AZURE, or GCP.
   * - -r, --region
     - string
     - false

>      - Physical location of your MongoDB cluster. For a complete list of supported AWS regions, see: https://dochub.mongodb.org/core/aws-atlas. For a complete list of supported Azure regions, see: https://dochub.mongodb.org/core/azure-atlas. For a complete list of supported GCP regions, see: https://dochub.mongodb.org/core/gcp-atlas..

Reviewer: ```suggestion
     - Physical location of your MongoDB deployment. For a complete list of supported AWS regions, see: https://dochub.mongodb.org/core/aws-atlas. For a complete list of supported Azure regions, see: https://dochub.mongodb.org/core/azure-atlas. For a complete list of supported GCP regions, see: https://dochub.mongodb.org/core/gcp-atlas..
```

Author: done

>      - Physical location of your MongoDB cluster. For a complete list of supported AWS regions, see: https://www.mongodb.com/docs/atlas/reference/amazon-aws/. For a complete list of supported Azure regions, see: https://www.mongodb.com/docs/atlas/reference/microsoft-azure/#microsoft-azure. For a complete list of supported GCP regions, see: https://www.mongodb.com/docs/atlas/reference/google-gcp/..

Reviewer: We should use dochub URLs.

```suggestion
     - Physical location of your MongoDB cluster. For a complete list of supported AWS regions, see: https://dochub.mongodb.org/core/aws-atlas. For a complete list of supported Azure regions, see: https://dochub.mongodb.org/core/azure-atlas. For a complete list of supported GCP regions, see: https://dochub.mongodb.org/core/gcp-atlas.
```

Author: done, thank you

   * - --skipMongosh
     - 
     - false
     - Indicates whether to skip accessing your deployment with MongoDB Shell.
   * - --skipSampleData
     - 
     - false

>      - Indicates whether to skip loading sample data into your MongoDB cluster.

Reviewer: ```suggestion
     - Indicates whether to skip loading sample data into your MongoDB deployment.
```

Author: done

>      - Indicates whether to skip loading sample data into your Atlas cluster.

Reviewer: ```suggestion
     - Indicates whether to skip loading sample data into your MongoDB cluster.
```

Author: done, thank you

   * - --tag
     - stringToString
     - false

>      - List that contains key-value pairs between 1 to 255 characters in length for tagging and categorizing the cluster. This value defaults to [].

Reviewer: ```suggestion
     - List that contains key-value pairs between 1 to 255 characters in length for tagging and categorizing the deployment. This value defaults to [].
```

Author: done

   * - --tier
     - string
     - false

>      - Tier for each data-bearing server in the cluster. To learn more about cluster tiers, see https://www.mongodb.com/docs/atlas/manage-clusters/#select-cluster-tier. This value defaults to "M0".

Reviewer: ```suggestion
     - Tier for each data-bearing server in the deployment. To learn more about cluster tiers, see https://dochub.mongodb.org/core/cluster-tier-atlas. This value defaults to "M0".
```

Author: done

.. _atlas-deployments-setup:

=======================
atlas deployments setup
=======================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

Create a local deployment.

Syntax
------

.. code-block::
   :caption: Command Syntax

   atlas deployments setup [deploymentName] [options]

.. Code end marker, please don't delete this comment

Arguments
---------

.. list-table::
   :header-rows: 1
   :widths: 20 10 10 60

   * - Name
     - Type
     - Required
     - Description
   * - deploymentName
     - string
     - false

>      - Name of the deployment you want to setup.

Reviewer: ```suggestion
     - Name of the deployment that you want to setup.
```

Author: done, thanks

Options
-------

.. list-table::
   :header-rows: 1
   :widths: 20 10 10 60

   * - Name
     - Type
     - Required
     - Description
   * - --connectWith
     - string
     - false
     - Method of connection. Valid values are mongosh, compass and skip.
   * - --force
     - 
     - false
     - Flag that indicates whether to skip the confirmation prompt before proceeding with the requested action.
   * - -h, --help
     - 
     - false

>      - help for setup

Reviewer: ```suggestion
     - Help for setup.
```

Author: same as https://github.com/mongodb/mongodb-atlas-cli/pull/2263/files#r1329411943
